GRDB/Record/PersistableRecord+Encodable.swift

```diff
@@ -40,7 +42,21 @@ private struct PersistableRecordKeyedEncodingContainer<Key: CodingKey> : KeyedEn
             // This allows us to encode Date as String, for example.
             encode((value as! DatabaseValueConvertible), key.stringValue)
         } else {
-            try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            do {
+                 try value.encode(to: PersistableRecordEncoder(codingPath: [key], encode: encode))
+            } catch {
+                // If value.encode does not work e.g. "unkeyed encoding is not supported" then see if model can be stored as JSON
+                let encodeError = error
+                do {
+                    let json = try JSONEncoder().encode(value)
```

Comment:
Please set encoder's outputFormatting to `.sortedKeys`. This will guarantee stable JSON output, and will ease record comparison.

Reply:
I'd also like that we think about which dateEncodingStrategy we should use. I don't quite like that we use the default format, which is not portable at all to other platforms. I'm thinking about people that share databases with Android.

Reply:
We were thinking of the same thing but .sortedKeys is iOS 11 only

Reply:
And make this point clear in the documentation (that record comparison may give odd results for records stored before iOS 11)

Reply:
Precisely speaking: iOS 11.0+, macOS 10.13+, watchOS 4.0+